vassal-app/src/main/java/VASSAL/build/module/Map.java

```diff
@@ -3063,6 +3064,21 @@ public void centerAt(Point p, int dx, int dy) {
     }
   }
 
+  /**
+   * @return the current visible rect of our map
+   */
+  public Rectangle getVisibleRect() {
+    return theMap.getVisibleRect();
+  }
+
+  /**
+   * @return the point our map's view is currently centered on
+   */
+  public Point getCenter() {
+    final Rectangle r = getVisibleRect();
+    return new Point(r.x + r.width/2, r.y + r.height/2);
+  }
```

Comment:
I'm surprised there wasn't something to do this already.

Reply:
Haha me too, cause it's one of the first things I write in a new videogame!